contracts/src/v0.8/l2ep/dev/scroll/ScrollSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {AggregatorInterface} from "../../../shared/interfaces/AggregatorInterface.sol";
+import {AggregatorV3Interface} from "../../../shared/interfaces/AggregatorV3Interface.sol";
+import {AggregatorV2V3Interface} from "../../../shared/interfaces/AggregatorV2V3Interface.sol";
+import {TypeAndVersionInterface} from "../../../interfaces/TypeAndVersionInterface.sol";
+import {ScrollSequencerUptimeFeedInterface} from "../interfaces/ScrollSequencerUptimeFeedInterface.sol";
+import {SimpleReadAccessController} from "../../../shared/access/SimpleReadAccessController.sol";
+import {IL2ScrollMessenger} from "@scroll-tech/contracts/L2/IL2ScrollMessenger.sol";
+
+///
+/// @title ScrollSequencerUptimeFeed - L2 sequencer uptime status aggregator
+/// @notice L2 contract that receives status updates, and records a new answer if the status changed
+///
+contract ScrollSequencerUptimeFeed is
+  AggregatorV2V3Interface,
+  ScrollSequencerUptimeFeedInterface,
+  TypeAndVersionInterface,
+  SimpleReadAccessController
+{
+  /// @dev Round info (for uptime history)
+  struct Round {
+    bool status;
+    uint64 startedAt;
+    uint64 updatedAt;
+  }
+
+  /// @dev Packed state struct to save sloads
+  struct FeedState {
+    uint80 latestRoundId;
+    bool latestStatus;
+    uint64 startedAt;
+    uint64 updatedAt;
+  }
+
+  /// @notice Sender is not the L2 messenger
+  error InvalidSender();
+  /// @notice Replacement for AggregatorV3Interface "No data present"
+  error NoDataPresent();
+
+  event L1SenderTransferred(address indexed from, address indexed to);
+  /// @dev Emitted when an `updateStatus` call is ignored due to unchanged status or stale timestamp
+  event UpdateIgnored(bool latestStatus, uint64 latestTimestamp, bool incomingStatus, uint64 incomingTimestamp);
+  /// @dev Emitted when a updateStatus is called without the status changing
+  event RoundUpdated(int256 status, uint64 updatedAt);
+
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  uint8 public constant override decimals = 0;
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  string public constant override description = "L2 Sequencer Uptime Status Feed";
```

Comment:
Why does this have a description?

Reply:
To maintain consistency with both [Arbitrum](https://github.com/smartcontractkit/chainlink/blob/67861738903ddfd7982c8bb9067ee1088874f713/contracts/src/v0.8/l2ep/dev/arbitrum/ArbitrumSequencerUptimeFeed.sol#L59) and [Optimism](https://github.com/smartcontractkit/chainlink/blob/67861738903ddfd7982c8bb9067ee1088874f713/contracts/src/v0.8/l2ep/dev/optimism/OptimismSequencerUptimeFeed.sol#L52)

Reply:
Wondering if it is actually used. Can't see any use of this over typeAndVersion

Reply:
Also wondering the same thing 🤔  I figured that it may be best to keep this around for now since the other two contracts have it, but I'm open to removing it in case it isn't needed

Reply:
Let's keep it around for now and see in the follow up PR(s)